File: 520-detect-capital/detect-capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        length = len(word)
        count = 0

        if length == 1:
            return True

        for ch in word:
            if ch.isupper():
                count += 1

        if count == 1 and word[0].isupper():

            return True
        if count == 0 or count == length:
            return True
        else:
            return False

File: 520-detect-capital/detect-capital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int len = word.size();
        int cnt = 0;
        if (len == 1) {
            return true;
        }
        for (int i = 0; i < len; i++) {
            if (isupper(word[i])) {
                cnt++;
            }
        }
        if (cnt == 1 && isupper(word[0])) {
            return true;
        }//if everything is capital then also true , if everything is small also true 
        if(cnt==0 || cnt==len)
        {
            return true;
        }
        else return false;

    }
};